crypto/ec/ecp_smpl.c

```diff
@@ -1363,3 +1365,57 @@ int ec_GFp_simple_field_sqr(const EC_GROUP *group, BIGNUM *r, const BIGNUM *a,
 {
     return BN_mod_sqr(r, a, group->field, ctx);
 }
+
+/*-
+ * Apply randomization of EC point projective coordinates:
+ *
+ *   (X, Y ,Z ) = (lambda^2*X, lambda^3*Y, lambda*Z)
+ *   lambda = [1,group->field)
+ *
+ */
+int ec_GFp_simple_blind_coordinates(const EC_GROUP *group, EC_POINT *p,
+                                    BN_CTX *ctx)
+{
+    int ret = 0;
+    BIGNUM *lambda = NULL;
+    BIGNUM *temp = NULL;
+
+    BN_CTX_start(ctx);
+    lambda = BN_CTX_get(ctx);
+    temp = BN_CTX_get(ctx);
+    if (temp == NULL) {
```

Comment:
And it's ok if lambda is NULL? I mean I suppose it should be `if (temp == NULL || lambda == NULL)`. No?

Reply:
Actually, no - it is correct as written. We have *lots* of code written like this. Successive calls to `BN_CTX_get()` are guaranteed to return NULL if any previous one has also returned NULL (this is a documented property).

Reply:
Cool.